src/exercise/dist/Main.js

```diff
@@ -0,0 +1,183 @@
+        const task_text = document.createElement("p");
+        task_text.innerText = value;
+        const task_elem = document.createElement("div");
+        task_elem.className = "tasks";
+        task_elem.onclick = () => alert("Task: \n" + task_text.innerText);
+        task_elem.appendChild(task_text);
+        task_elem.appendChild(this.taskTime(date));
+        return task_elem;
+    };
+
+    createDeleteBtn(task_section){
+        const delete_btn = document.createElement("button");
+        delete_btn.className = "deleteBtn";
+        delete_btn.onclick = () => {
+            task_section.classList.add("removed");
+            setTimeout(() => {
+                task_section.remove();
+                const task_date = task_section.querySelectorAll('p');
+                this.itemClient.deleteTask(task_date[0].innerText, task_date[1].id);
+            }, 800);
+            return;
+        }
+        const deleteIcon = document.createElement("i");
+        deleteIcon.className = "fa fa-trash-o";
+        delete_btn.appendChild(deleteIcon);
+        return delete_btn;
+    };
+
+
+    async renderTasks(){
+        // clear all tasks
+        document.body.querySelector("div.taskList").innerHTML = '';        
+        // render tasks
+        const tasks = await this.itemClient.getTasks();
+        tasks.data.forEach(item => {
+            // create div for task + delete button
+            const task_section = document.createElement("div");
+            task_section.className = "taskSection";
+            document.body.querySelector("div.taskList").appendChild(task_section);
+            // create the task    
+            task_section.appendChild(this.createTask(item.task, item.date));
+            // create delete button
+            task_section.appendChild(this.createDeleteBtn(task_section));
+        });
+    };
+
+    async addTask(task) {
+        if (task.value === ''){
+            task.setCustomValidity("You forgot to type the task!");
+            task.reportValidity();
+            return;
+        }
+        const loader = document.getElementsByClassName("loader")[0];
+        loader.style.display = 'block';
+        if (task.value) {
+            try{
+                await this.itemClient.addTask(task.value, new Date());
+            } catch(error) {
+                loader.style.display = 'none';
+                task.setCustomValidity(error.message);
+                task.reportValidity();
+                return;
+            }
+        }
+        loader.style.display = 'none';
+        await this.renderTasks();
+        task.value = '';
+    }
+
+    // returns the element that shows the date & time the task created
+    taskTime(dateString){
+        let date = new Date(dateString);
+        const p = document.createElement("p");
+        let minutes = String(date.getMinutes());
+        if (minutes.length === 1){
+            minutes = "0" + minutes;
+        }
+        p.innerHTML = date.getDate() + "/" + 
```

Comment:
For readability:
1. export a formatDate function
2. use the `${}` syntax for string manipulation instead of + + +